File: Domain/ValueObjects/LeavePeriodDTO.cs
using Domain.Agregates;
using Domain.Entities;
using Domain.Entities.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.ValueObjects
{
    public record LeavePeriodDTO(
        string EmployeeId,
        LeavePeriodDemandDTO? LeavePeriodDemand
        )
    {
        public LeavePeriod ToDomain()
        {
            return new LeavePeriod()
            {
                Employee = new Employee(EmployeeId),
                LeavePeriodDemand = LeavePeriodDemand?.ToDomain(),
            };
        }
    }
}
    

File: tf1testech.client/src/models/LeavePeriodDTO.ts
import LeavePeriodDemandDTO from "./LeavePeriodDemandDTO";

class LeavePeriodDTO {
    employeeId: string;
    leavePeriodDemand: LeavePeriodDemandDTO;

    constructor(employeeId: string, leavePeriodDemandDTO: LeavePeriodDemandDTO){
        this.employeeId = employeeId;
        this.leavePeriodDemand = leavePeriodDemandDTO;
    }
}

export default LeavePeriodDTO;